transport/src/main/java/io/netty/channel/socket/TunPacket.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.socket;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.DefaultByteBufHolder;
+
+import java.net.InetAddress;
+
+/**
+ * Envelope class for IPv4 and IPv6 packets received from/sent to TUN devices.
+ *
+ * @see Tun4Packet
+ * @see Tun6Packet
+ */
+public abstract class TunPacket extends DefaultByteBufHolder {
+    protected TunPacket(ByteBuf data) {
+        super(data);
+    }
+
+    /**
+     * Returns the IP version.
+     *
+     * @return the IP version.
+     */
+    public abstract int version();
```

Comment:
How about using an `enum`?

Reply:
I can add this enum that I made for another project: https://github.com/drasyl-overlay/netty-tun/blob/master/src/main/java/org/drasyl/channel/tun/InetProtocol.java

Reply:
That looks good.

Reply:
As this method returns the IP version, not a layer 4 protocol number. Therefore, I changed the method to return a [`InternetProtocolFamily`](https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/socket/InternetProtocolFamily.java) enum.